pathing.py

```diff
@@ -45,12 +46,38 @@ def set_movement(target) -> bool:
         return False
 
 
-def approach_actor(actor_id: int = 999, actor_index: int = 999, talk: bool = True):
+def distance(actor_index:int):
+    # Assume index is passed in.
+    actor_coords = memory.main.get_actor_coords(actor_index=actor_index)
+    player_coords = memory.main.get_coords()
+    distance = sqrt(
+        ((player_coords[0] - actor_coords[0])**2) +
+        ((player_coords[1] - actor_coords[1])**2)
+    )
+    return int(distance)
+
+def approach_actor(actor_id:int=999, actor_index:int=999, talk:bool=True):
```

Comment:
A few things here.

* Traditionally when we are putting in optional arguments we would do `actor_id=None`. My money says that there are enough actors that it would be possible for one of them to have an id of 999. This gets around that.
* I think it would be better to split this into two or three functions:
  * def approach_actor_by_id(actor_id:int, talk:bool=True):
  * def approach_actor_by_index(actor_index:int, talk:bool=True):
  * def approach_player(player:Player, talk:bool=True):
* It looks like at the end of the day, we care about the `actor_index`. So `approch_actor_by_index` should do all the work, and the other two can just call that one by getting the `actor_index` then passing that along.

This does a few things. No more need for checking arguments to make sure only one was passed in. We are explicitly saying which one we are expecting. It cleans up some of the logic in here that is checking to make sure we are using this function correctly. There are no more chances to use it incorrectly. If we want to talk more about this change before taking a stab at it, please let me know either here or in discord. 

This is mostly a clean-up/refactor request. Best time do to those are when we are tinkering with them already.

Reply:
Refactored with three more read-able functions that pass through to the primary engine for this feature.